[WIP] Adds rbac proxy sidecar container to authorinos instance

jjaferson opened the pull request:
Adds rbac sidecar proxy to protect authorinos metric endpoint

Closes: https://github.com/Kuadrant/authorino-operator/issues/4

eguzki commented:
I am not sure whether this is convenient. 

We removed them from the 3scale operator https://github.com/3scale/3scale-operator/pull/692 

Ops team has also removed the rbac proxy from their operator https://github.com/3scale-ops/prometheus-exporter-operator/pull/26

Check out for the reasons in the PRs

jjaferson commented:
> I am not sure whether this is convenient.
> 
> We removed them from the 3scale operator [3scale/3scale-operator#692](https://github.com/3scale/3scale-operator/pull/692)
> 
> Ops team has also removed the rbac proxy from their operator [3scale-ops/prometheus-exporter-operator#26](https://github.com/3scale-ops/prometheus-exporter-operator/pull/26)
> 
> Check out for the reasons in the PRs

Makes sense @eguzki I'll closing this PR and we will remove the ClusterRole for the rbac-proxy container in authorino.
